tests/cxx/pragma_keep_multi.cc

```diff
@@ -0,0 +1,43 @@
+//===--- pragma_keep_multi.cc - test input file for iwyu ------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// IWYU_ARGS: -I .
+
+// This tests whether or not begin_keep/end_keep works
+
+// IWYU pragma: begin_keep
```

Comment:
I would like to see a few more variations:

* Empty `begin_keep`/`end_keep` block
* `begin_keep`/`end_keep` block with a single include line
* `begin_keep`/`end_keep` block with multiple include lines
* `begin_keep`/`end_keep` block with a single forward declaration
* `begin_keep`/`end_keep` block with multiple forward declarations
* `begin_keep`/`end_keep` block with include lines and forward declarations mixed

For the most part you can just regroup the existing includes, but you may need some new forward declarations.